pytest_inmanta/plugin.py

```diff
@@ -951,6 +951,39 @@ def dryrun_resource(
         assert ctx.status == status
         return ctx.changes
 
+    def dryrun_all(
+        self, run_as_root: bool = False, assert_create_or_delete: bool = False
+    ) -> Dict[Resource, HandlerContext]:
+        """
+        Runs a dryrun for every resource.
```

Comment:
This only returns resources with changes. I can't comment on the correctness of that, but could you add it to the docstring so it's clear to the user?

Reply:
I changed it to return everything. 

> I think the tests could be expanded a bit:
> 
>     * assert the method returns the expected value
> 
>     * add some failure scenarios (if feasible): what if the deploy fails? What if the dryrun after the deploy is not empty?

Regarding this comment. I tried to do more thorough tests but I couldn't actually get the dryrun to succeed and the deploy to fail. The dryrun after the deploy I assume is only empty if something went wrong with the deploy. Do you have any idea on how to test this?

Reply:
Could you still use help with this?